Userland/Libraries/LibJS/Runtime/Temporal/ISO8601.cpp

```diff
@@ -0,0 +1,505 @@
+/*
+ * Copyright (c) 2021, Linus Groh <linusg@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/CharacterTypes.h>
+#include <LibJS/Runtime/Temporal/ISO8601.h>
+
+namespace JS::Temporal {
+
+namespace Detail {
+
+#define TRY_PARSE(scope)                            \
```

Comment:
This isn't something I'm very involved with, but...can we calm down with the macros?
This can be done with less macro and more C++ (see next comment for an example)

Reply:
Done, tyvm for giving some examples! This was a pain to refactor across commits, but I like it much better.